oop_rock_paper_scissors.py

```diff
@@ -0,0 +1,45 @@
+""" Object-Oriented Rock Paper Scissors """
+
+import random
+
+
+class User(object):
+
+    user_score = 0
+    computer_score = 0
+
+    valid_choices = ["rock", "paper", "scissors"]
+
+    def __init__(self):
+        self.choice = ''
+
+    # prompt user input
+    def prompt_move(self):
+        user_choice = input("Choose rock, paper, or scissors: ")
+        self.choice = user_choice
+
+    # return user move
+    def move(self):
+        return self.choice
+
+
+class NPCUser(User):
+
+    # get random computer move
+    def move(self):
```

Comment:
This is an EXCELLENT use of inheritance! I was hoping this would be the route you went with. I wonder if we can't structure this a little more similarly to the parent class so the API is more standard. The workflow for human user seems to be: `user.prompt_move()` and `user.move()` whereas an NPCUser is just `user.move()`. Maybe `prompt_move` can be abstracted a bit and renamed `get_move`, which would prompt a human user and call `random.choice` for an NCP. Then the workflow for both would be the same: `user.get_move` and `user.move`. Now you're respecting the concept that methods should do one, and only one, thing (`get_move` gets the move and `move` plays the move (although I might consider renaming that to `play_move` so it's not ambiguous whether that is a noun or a verb)).